The CHANGELOG. (#3089)
mlpack/mlpack: HISTORY.md

### mlpack ?.?.?
###### ????-??-??
  * Migrate from boost tests to Catch2 framework (#2523), (#2584).

  * Bump minimum armadillo version from 8.400 to 9.800 (#3043), (#3048).

  * Adding a copy constructor in the Convolution layer (#3067).

>   * Replace boost::spirit parser by a local efficient implementation (#2942).

Reviewer: ```suggestion
  * Replace `boost::spirit` parser by a local efficient implementation (#2942).
```

  * Disable correctly the autodownloader + fix tests stability (#3076).

>   * Replace boost::any with core::v2::any or std::any if available (#3006).

Reviewer: ```suggestion
  * Replace `boost::any` with `core::v2::any` or `std::any` if available (#3006).
```

  * Remove old non used Boost headers (#3005).

>   * Replace boost::enable_if with std::enable_if (#2998).

Reviewer: ```suggestion
  * Replace `boost::enable_if` with `std::enable_if` (#2998).
```

>   * Replace boost::is_same with std::is_same (#2993).

Reviewer: ```suggestion
  * Replace `boost::is_same` with `std::is_same` (#2993).
```

  * Remove invalid option for emsmallen and STB (#2960).

  * Check for armadillo dependencies before downloading armadillo (#2954).

  * Disable the usage of autodownloader by default (#2953).

  * Install dependencies downloaded with the autodownloader (#2952).
 
  * Download older Boost if the compiler is old (#2940). 

  * Add support for embedded systems (#2531).

  * Build mlpack executable statically if the library is statically linked (#2931).

  * Fix cover tree loop bug on embedded arm systems (#2869).

>   * Fix a LAPACK bug in FindArmadillo.cmake (#2929).

Reviewer: ```suggestion
  * Fix a LAPACK bug in `FindArmadillo.cmake` (#2929).
```

### mlpack ?.?.?
###### ????-??-??
  * Migrate from boost tests to Catch2 framework (#2523), (#2584).

  * Update NumFOCUS team voting responsibilities (#3071).

  * Add armadillo bump version policy (#3048).

  * Bump minimum armadillo version from 8.400 to 9.800 (#3043).

  * Adding a copy constructor in the Convolution layer (#3067).

  * Replace boost::spirit parser by a local efficient implementation
   (#2942).

  * Disable correctly the autodownloader + fix tests stability (#3076).

>   * Replace boost::any with core::v2::any (#3006).

Reviewer 2: ```suggestion
  * Replace boost::any with core::v2::any or std::any if available
    (#3006).
```
I think that is correct?  Feel free to ignore the suggestion if it's not.

Author: Yes it is valid for Windows

>   * Remove old non used Boost headers (#3005)

Reviewer 2: ```suggestion
  * Remove old non used Boost headers (#3005).
```
Sorry for the pedantry. :)

  * Replace boost::enable_if with std::enable_if (#2998).

  * Replace boost::is_same with std::is_same (#2993).

>   * Remove non valid option for emsmallen and stb (#2960).

Reviewer 2: ```suggestion
  * Remove invalid option for emsmallen and STB (#2960).
```

>   * Check for armadillo dependecies before downloading armadillo (#2954).

Reviewer 2: ```suggestion
  * Check for armadillo dependencies before downloading armadillo (#2954).
```

  * Disable the usage of autodownloader by default (#2953).

>   * Install dependecies downloaded with the autodownloader (#2952).

Reviewer 2: ```suggestion
  * Install dependencies downloaded with the autodownloader (#2952).
```

### mlpack ?.?.?
###### ????-??-??
  * Migrate from boost tests to Catch2 framework (#2523), (#2584).

>   * Update NumFOCUS team voting responsibilities (#3071).

Reviewer: I would only list code/documentation changes, I'm not even sure this file is part or should be part of a release.

  * Add armadillo bump version policy (#3048).

>   * Bump minimum armadillo version from 8.400 to 9.800 (#3043).

Reviewer: I would move `#3048` into the `Bump minimum armadillo version from 8.400 to 9.800 (#3043).`, similar reason as mentioned above.

  * Adding a copy constructor in the Convolution layer (#3067).

>   * Replace boost::spirit parser by a local efficient implementation
>    (#2942).

Reviewer: ```suggestion
  * Replace boost::spirit parser by a local efficient implementation (#2942).
```

I don't think we should use the 80 character rule on the history file.

  * Disable correctly the autodownloader + fix tests stability (#3076).

>   * Replace boost::any with core::v2::any or std::any if available
>     (#3006).

Reviewer: ```suggestion
  * Replace boost::any with core::v2::any or std::any if available (#3006).
```

See comment above.

  * Remove old non used Boost headers (#3005).

  * Replace boost::enable_if with std::enable_if (#2998).

  * Replace boost::is_same with std::is_same (#2993).

  * Remove invalid option for emsmallen and STB (#2960).

  * Check for armadillo dependencies before downloading armadillo (#2954).

  * Disable the usage of autodownloader by default (#2953).

  * Install dependencies downloaded with the autodownloader (#2952).
 
  * Download older Boost if the compiler is old (#2940). 

  * Add support for embedded systems (#2531).

>   * Build mlpack executable statically if the library is statically linked
>    (#2931).

Reviewer: ```suggestion
  * Build mlpack executable statically if the library is statically linked (#2931).
```

See comment above.

  * Fix cover tree loop bug on embedded arm systems (#2869).

  * Fix a LAPACK bug in FindArmadillo.cmake (#2929).

>   * Fix Boost installation on the CI (#2928).

Reviewer: I would remove that one, for the reasons mentioned in the first comment.

Author: Okay, so basically remove all lines related to CI, if I understand well.

Reviewer: At least that's what I would do, but maybe others like to list everything, @rcurtin what do you think?

Author: no worries, I have already removed them,
I think in the future we will have everything listed if things becomes automatic.

  * Add an autodownloader to get mlpack dependencies (#2927).

  * Remove Coverage files and configurations from CMakeLists (#2866).

>   * Remove the double installation of Armadillo in the CI (#2837).

Reviewer: I would remove that one, for the reasons mentioned in the first comment.
